homeassistant/components/proxmoxve/config_flow.py

```diff
@@ -0,0 +1,186 @@
+
+    def __init__(self):
+        """Initialize config flow."""
+        super().__init__()
+
+        self._config = {}
+        self.proxmox_client = None
+
+    # pylint: disable=signature-differs
+    async def async_step_user(self, user_input):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            self._config[CONF_HOST] = user_input.get(CONF_HOST)
+            self._config[CONF_PORT] = user_input.get(CONF_PORT)
+            self._config[CONF_USERNAME] = user_input.get(CONF_USERNAME)
+            self._config[CONF_PASSWORD] = user_input.get(CONF_PASSWORD)
+            self._config[CONF_REALM] = user_input.get(CONF_REALM)
+            self._config[CONF_VERIFY_SSL] = user_input.get(CONF_VERIFY_SSL)
+
+            self.proxmox_client = ProxmoxClient(
+                self._config[CONF_HOST],
+                self._config[CONF_PORT],
+                self._config[CONF_USERNAME],
+                self._config[CONF_REALM],
+                self._config[CONF_PASSWORD],
+                self._config[CONF_VERIFY_SSL],
+            )
+
+            try:
+                await self.hass.async_add_executor_job(self.proxmox_client.build_client)
+            except AuthenticationError:
+                errors["base"] = "auth_error"
+            except SSLError:
+                errors["base"] = "ssl_error"
+            except (ConnectTimeout, RConnectionError):
+                errors["base"] = "server_unreachable"
+            except Exception:  # pylint: disable=broad-except
+                errors["base"] = "generic_error"
+                _LOGGER.exception("ProxmoxVE generic error")
+
+            if not errors:
+                _LOGGER.info("ProxmoxVE login successful, go to next page")
+                return await self.async_step_select_nodes_vms(None)
+
+        if user_input is None:
+            user_input = {}
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+                vol.Required(
+                    CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")
+                ): str,
+                vol.Required(
+                    CONF_PASSWORD, default=user_input.get(CONF_PASSWORD, "")
+                ): str,
+                vol.Required(
+                    CONF_REALM, default=user_input.get(CONF_REALM, DEFAULT_REALM)
+                ): str,
+                vol.Optional(
+                    CONF_PORT, default=user_input.get(CONF_PORT, DEFAULT_PORT)
+                ): int,
+                vol.Optional(
+                    CONF_VERIFY_SSL,
+                    default=user_input.get(CONF_VERIFY_SSL, DEFAULT_VERIFY_SSL),
+                ): bool,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors,
+        )
+
+    async def async_step_select_nodes_vms(self, user_input):
```

Comment:
We should add all VMs and add this selection.
When adding a device/integration it should add everything. If there are less common sensors, one can decide to disable those by default.

Reply:
What do you mean by that?

We should add all VMs and Container to home assistant automatically?

If I have understood correctly, I should add all VMs and Containers with a unique id and the user can disable the binary sensors that does not want from home assistant itself.

How do we handle moving the old configuration though?
You could decide which VMs to add in the old config, that piece of config would be lost moving to the config flow